In [2]:
from classify import *

In [3]:
create_vocabulary('./EasyFiles/', 1)

[',',
 '.',
 '19',
 '2020',
 'a',
 'cat',
 'chases',
 'dog',
 'february',
 'hello',
 'is',
 'it',
 'world']

Test 1: [',', '.', '19', '2020', 'a', 'cat', 'chases', 'dog', 'february', 'hello', 'is', 'it', 'world']

In [4]:
create_vocabulary('./EasyFiles/', 2)

['.', 'a']

Test 2: ['.', 'a']

In [5]:
vocab = create_vocabulary('./EasyFiles/', 1)
create_bow(vocab, './EasyFiles/2016/1.txt')

{'.': 1, 'a': 2, 'cat': 1, 'chases': 1, 'dog': 1}

Test 3: {'a': 2, 'dog': 1, 'chases': 1, 'cat': 1, '.': 1}

In [6]:
create_bow(vocab, './EasyFiles/2020/2.txt')

{',': 1, '.': 1, '19': 1, '2020': 1, 'february': 1, 'is': 1, 'it': 1}

Test 4: {'it': 1, 'is': 1, 'february': 1, '19': 1, ',': 1, '2020': 1, '.': 1}

In [7]:
vocab = create_vocabulary('./EasyFiles/', 2)
create_bow(vocab, './EasyFiles/2016/1.txt')

{'.': 1, 'a': 2, None: 3}

Test 5: {'a': 2, None: 3, '.': 1}

In [8]:
vocab = create_vocabulary('./EasyFiles/', 1)
load_training_data(vocab,'./EasyFiles/')

[{'label': '2016', 'bow': {'hello': 1, 'world': 1}},
 {'label': '2016', 'bow': {'.': 1, 'a': 2, 'cat': 1, 'chases': 1, 'dog': 1}},
 {'label': '2020',
  'bow': {',': 1,
   '.': 1,
   '19': 1,
   '2020': 1,
   'february': 1,
   'is': 1,
   'it': 1}}]

Test 6: [{'label': '2020', 'bow': {'it': 1, 'is': 1, 'february': 1, '19': 1, ',': 1, '2020': 1, '.': 1}},
    {'label': '2016', 'bow': {'hello': 1, 'world': 1}},
    {'label': '2016', 'bow': {'a': 2, 'dog': 1, 'chases': 1, 'cat': 1, '.': 1}}]

In [9]:
vocab = create_vocabulary('./corpus/training/', 2)
training_data = load_training_data(vocab,'./corpus/training/')
prior(training_data, ['2020', '2016'])

{'2020': -0.3217118210380923, '2016': -1.2906462863976693}

Test 7: {'2020': -0.32171182103809226, '2016': -1.2906462863976689}

In [10]:
vocab = create_vocabulary('./EasyFiles/', 1)
training_data = load_training_data(vocab, './EasyFiles/')
p_word_given_label(vocab, training_data, '2020')

{',': -2.3513752571634776,
 '.': -2.3513752571634776,
 '19': -2.3513752571634776,
 '2020': -2.3513752571634776,
 'a': -3.044522437723423,
 'cat': -3.044522437723423,
 'chases': -3.044522437723423,
 'dog': -3.044522437723423,
 'february': -2.3513752571634776,
 'hello': -3.044522437723423,
 'is': -2.3513752571634776,
 'it': -2.3513752571634776,
 'world': -3.044522437723423,
 None: -3.044522437723423}

Test 8: {',': -2.3513752571634776, '.': -2.3513752571634776, '19': -2.3513752571634776, '2020': -2.3513752571634776, 'a': -3.044522437723423, 'cat': -3.044522437723423, 'chases': -3.044522437723423, 'dog': -3.044522437723423, 'february': -2.3513752571634776, 'hello': -3.044522437723423, 'is': -2.3513752571634776, 'it': -2.3513752571634776, 'world': -3.044522437723423, None: -3.044522437723423}

In [11]:
p_word_given_label(vocab, training_data, '2016')

{',': -3.0910424533583156,
 '.': -2.3978952727983707,
 '19': -3.0910424533583156,
 '2020': -3.0910424533583156,
 'a': -1.9924301646902063,
 'cat': -2.3978952727983707,
 'chases': -2.3978952727983707,
 'dog': -2.3978952727983707,
 'february': -3.0910424533583156,
 'hello': -2.3978952727983707,
 'is': -3.0910424533583156,
 'it': -3.0910424533583156,
 'world': -2.3978952727983707,
 None: -3.0910424533583156}

Test 9: {',': -3.091042453358316, '.': -2.3978952727983707, '19': -3.091042453358316, '2020': -3.091042453358316, 'a': -1.9924301646902063, 'cat': -2.3978952727983707, 'chases': -2.3978952727983707, 'dog': -2.3978952727983707, 'february': -3.091042453358316, 'hello': -2.3978952727983707, 'is': -3.091042453358316, 'it': -3.091042453358316, 'world': -2.3978952727983707, None: -3.091042453358316}

In [12]:
vocab = create_vocabulary('./EasyFiles/', 2)
training_data = load_training_data(vocab, './EasyFiles/')
p_word_given_label(vocab, training_data, '2020')

{'.': -1.6094379124341003,
 'a': -2.3025850929940455,
 None: -0.35667494393873245}

Test 10: {'.': -1.6094379124341005, 'a': -2.302585092994046, None: -0.35667494393873267}

In [13]:
p_word_given_label(vocab, training_data, '2016')

{'.': -1.7047480922384253, 'a': -1.2992829841302609, None: -0.6061358035703156}

Test 11: {'.': -1.7047480922384253, 'a': -1.2992829841302609, None: -0.6061358035703157}

In [14]:
train('./EasyFiles/', 2)

{'vocabulary': ['.', 'a'],
 'log prior': {'2020': -0.916290731874155, '2016': -0.5108256237659907},
 'log p(w|y=2016)': {'.': -1.7047480922384253,
  'a': -1.2992829841302609,
  None: -0.6061358035703156},
 'log p(w|y=2020)': {'.': -1.6094379124341003,
  'a': -2.3025850929940455,
  None: -0.35667494393873245}}

Test 12: {'vocabulary': ['.', 'a'],
    'log prior': {'2020': -0.916290731874155, '2016': -0.5108256237659905},
    'log p(w|y=2020)': {'.': -1.6094379124341005, 'a': -2.302585092994046, None: -0.35667494393873267},
    'log p(w|y=2016)': {'.': -1.7047480922384253, 'a': -1.2992829841302609, None: -0.6061358035703157}}

In [15]:
model = train('./corpus/training/', 2)
classify(model, './corpus/test/2016/0.txt')

{'log p(y=2020|x)': -3906.351945884105,
 'log p(y=2016|x)': -3916.458747858925,
 'predicted y': '2020'}

Test 13: {'log p(y=2020|x)': -3906.351945884105, 'log p(y=2016|x)': -3916.458747858926, 'predicted y': '2020'}